### DISCUSSION TITLE
so-elasticsearch and so-kibana shard doesn't close automatically

### DISCUSSION THREAD

**Discussion by Ron89** [Security-Onion-Solutions/securityonion#8284] [Unsupported Versions] [answered]
I am using Security Onion 2.3. Starting few days ago, we start to miss alerts from components. After searching the log, we find out that the issue was "Validation Failed: 1: this action would add [5] shards, but this cluster currently has [1000]/[1000] maximum normal shards open;". We alleviated it by reducing the shards created by various components in `/opt/so/saltstack/local/pillar/global.sls:index_settings`, however, the issue keeps returned after a few days.

Today when checking the existing shards, we noticed that `so-elasticsearch` and `so-kinaba` shards are never ever closed or deleted. And this caused a accumulation of 1 shard for each every day. Over time, it simply overwhelmed the 1000 shards limit. From the official documentation, we do not see any instructions on configuring closing / deleting configurations on these two components anywhere. We also do not see implementation on their indices maintenance in `/opt/so/saltstack/default/salt/elasticsearch/templates/so/so-*-template.json.jinja`. Is it a missed out part of the implementation? Or is it intentional that `so-elasticsearch` and `so-kibana` shards shouldn't be closed?

Since this issue is essentially blinding us from new threats, we hope we can resolve it as fast as possible. Is it possible to manually close the `so-elasticsearch` and `so-kinana` shards? What would be the consequences if we do?

**Comment by dougburks** [comment, marked as answer]
> I am using Security Onion 2.3.

What exact version of 2.3 are you using?

Starting in Security Onion 2.3.80, `so-elasticsearch` indices should be closed after 30 days by default. Starting in the upcoming 2.3.140 release, `so-kibana` indices should also be closed after 30 days by default.

You could also check the curator logs to make sure curator is running as expected.

(UPDATE: Edited for clarity.)

**Comment by Ron89** [comment]
OK, we are using 2.3.70. Will use soup to update and see if it is resolved. Thanks.

**Comment by btdewey** [comment]
We are running 2.3.130 and do not have this issue.

**Reply by btdewey** [reply]
Err, I meant 2.3.120.  I saw this post so I wanted to make sure we didn't have the same issue.

**Reply by Ron89** [reply]
hmm, according to them, 2.3.130 solves the so-elasticsearch indices issue. We have done the upgrade and can confirm that the issue is resolved. For so-kibana, the issue is still there. We are making preparation to update to 2.3.140 (which is yet to release at the moment), as there are some Elasticsearch related warnings we need to handle before it is ready for ElasticSearch 8 (https://docs.securityonion.net/en/2.3/soup.html).

Note that the issue won't cause any trouble until you have used SO for around a year or so, because what this issue create is a slow cumulation of unclosed shards (2 per day). Since normal service requires around 400 shards regularly, and that the node configuration allow 1000 shards to be opened at the same time, you will only run into shard caused issues about 300 days or so in.

**Reply by dougburks** [reply]
I updated my reply above for clarity. I mentioned 2.3.130 as its the current release but the first main release that had a fix for the so-elasticsearch indices issue was 2.3.80.

**Comment by btdewey** [comment]
We looked at our shards and do not have any groups that are over our
retention that we can see.  What are you using to investigate this? I'm
interested in learning how your seeing this issue.

Bill

**Reply by Ron89** [reply]
@btdewey ,
To see it, go to Kibana, Management > Stack Management > Data > Index Management, search for `so-kinaba`.
Actually, as of 2.3.130, I see that both so-elasticsearch and so-kibana shards are closed regularly now. Is `so-kibana` issue resolved at 2.3.130 (  @dougburks  )?

**Reply by dougburks** [reply]
2.3.140 fixed an [issue](https://github.com/Security-Onion-Solutions/securityonion/issues/8262) where `so-kibana` indices were not being closed on search nodes. If you're running a standalone box, then `so-kibana` indices may be closed on 2.3.130.
